### DISCUSSION TITLE
Need help if Tine is in a subfolder with ActiveSync - Autodiscover

### DISCUSSION THREAD

**Discussion by gongoscho** [tine20/tine20#7258] [General]
Hello,

i need help if Tine is in a subfolder with the .htaccess file.

Tine runs on a web space from the german hoster: all-inkl.com without root access on PHP Version 7.4.10 as FPM/FastCGI.

I opened a ticket, but the hoster they can't really help me either.

The following config

domain.tld
/www/.htaccess
```
<IfModule mod_rewrite.c>
    RewriteEngine on
	
    # ActiveSync
    RewriteRule ^Microsoft-Server-ActiveSync /tine/index.php?frontend=activesync         [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
	
    # OpenID
    RewriteRule ^users/(.*)                  /tine/index.php?frontend=openid&username=$1 [L,QSA]

    # WebDAV / CalDAV / CardDAV
    RewriteCond %{REQUEST_METHOD} !^(GET|POST)$
    RewriteRule ^$                           /tine/index.php?frontend=webdav             [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]

    RewriteRule ^addressbooks                /tine/index.php?frontend=webdav             [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
    RewriteRule ^calendars                   /tine/index.php?frontend=webdav             [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
    RewriteRule ^webdav                      /tine/index.php?frontend=webdav             [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
    RewriteRule ^principals                  /tine/index.php?frontend=webdav             [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
    RewriteRule ^remote.php                  /tine/index.php?frontend=webdav             [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
    
    # Anonymous downloads
    RewriteRule ^download/get/(.*)           /tine/index.php?method=Download.downloadNode&path=$1 [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
    RewriteRule ^download/show/(.*)          /tine/index.php?method=Download.displayNode&path=$1  [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
	
	# Routing

	RewriteCond %{REQUEST_FILENAME} -s [OR]
	RewriteCond %{REQUEST_FILENAME} -l [OR]
	RewriteCond %{REQUEST_FILENAME} -d
	RewriteRule ^.*$ - [NC,L]
	RewriteRule ^.*$ /tine/index.php?doRouting=1 [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]
</IfModule>
```

/www/tine/.htaccess

```
DirectoryIndex index.php

<IfModule mod_php7.c>
    php_value max_input_time 120
    php_value session.gc_maxlifetime 86400
    php_value memory_limit 128M
    php_value upload_max_filesize 20M
    php_value post_max_size 20M
    php_value safe_mode off
    php_flag display_errors off
    php_flag log_errors on
    php_flag magic_quotes_gpc off
    php_flag magic_quotes_runtime off
    php_flag register_globals off
    php_value max_execution_time 90
</IfModule>
<IfModule mod_expires.c>
    ExpiresActive on

    ExpiresByType image/gif "access plus 1 month 1 days"
    ExpiresByType image/jpeg "access plus 1 month 1 days"
    ExpiresByType image/png "access plus 1 month 1 days" 
</IfModule>
```

ActiveSync is working: https://domain.tld/Microsoft-Server-ActiveSync return a "It works!"

So the paths in the rewrite rule are correct

Only Autodiscover does not: https://domain.tld/Autodiscover/Autodisover.xml

> An HTTP 500 response was returned from Unknown.
> 
> HTTP Response Headers:
> 
> X-API: http://www.tine20.org/apidocs/tine20/
> 
> Strict-Transport-Security: max-age=15552000
> 
> Upgrade: h2
> 
> Upgrade: h2c
> 
> Connection: Upgrade
> 
> Connection: close
> 
> Transfer-Encoding: chunked
> 
> Content-Type: text/html; charset=UTF-8
> 
> Date: Wed, 06 Jan 2021 10:10:09 GMT
> 
> Server: Apache
> 

In the Tine-Log ( Level: Warning (4) ) i see Errors:

> d961fd - -- none -- - 2021-01-06T10:10:10+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::133 ParseError -> syntax error, unexpected 'version' (T_STRING)
> d961fd - -- none -- - 2021-01-06T10:10:10+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::155 #0 .../vendor/zendframework/zendframework1/library/Zend/View/Abstract.php(888): Zend_View->_run('/www/htdocs/v14...')
> #1 .../Tinebase/Controller.php(1665): Zend_View_Abstract->render(NULL)
> #2 .../Tinebase/Expressive/RouteHandler.php(185): Tinebase_Controller->publicApiMSAutodiscoverXml()
> #3 .../Tinebase/Expressive/Middleware/Dispatch.php(45): Tinebase_Expressive_RouteHandler->dispatch()
> #4 .../vendor/zendframework/zend-stratigility/src/Next.php(60): Tinebase_Expressive_Middleware_Dispatch->process(Object(Zend\Diactoros\ServerRequest), Object(Zend\Stratigility\Next))
> #5 .../Tinebase/Expressive/Middleware/RoutePipeInject.php(63): Zend\Stratigility\Next->handle(Object(Zend\Diactoros\ServerRequest))
> #6 .../vendor/zendframework/zend-stratigility/src/Next.php(60): Tinebase_Expressive_Middleware_RoutePipeInject->process(Object(Zend\Diactoros\ServerRequest), Object(Zend\Stratigility\Next))
> #7 .../Tinebase/Expressive/Middleware/CheckRouteAuth.php(77): Zend\Stratigility\Next->handle(Object(Zend\Diactoros\ServerRequest))
> #8 .../vendor/zendframework/zend-stratigility/src/Next.php(60): Tinebase_Expressive_Middleware_CheckRouteAuth->process(Object(Zend\Diactoros\ServerRequest), Object(Zend\Stratigility\Next))
> #9 .../Tinebase/Expressive/Middleware/FastRoute.php(80): Zend\Stratigility\Next->handle(Object(Zend\Diactoros\ServerRequest))
> #10 .../vendor/zendframework/zend-stratigility/src/Next.php(60): Tinebase_Expressive_Middleware_FastRoute->process(Object(Zend\Diactoros\ServerRequest), Object(Zend\Stratigility\Next))
> #11 .../Tinebase/Expressive/Middleware/ResponseEnvelop.php(40): Zend\Stratigility\Next->handle(Object(Zend\Diactoros\ServerRequest))
> #12 .../vendor/zendframework/zend-stratigility/src/Next.php(60): Tinebase_Expressive_Middleware_ResponseEnvelop->process(Object(Zend\Diactoros\ServerRequest), Object(Zend\Stratigility\Next))
> #13 .../vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): Zend\Stratigility\Next->handle(Object(Zend\Diactoros\ServerRequest))
> #14 .../vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(72): Zend\Stratigility\MiddlewarePipe->process(Object(Zend\Diactoros\ServerRequest), Object(Zend\Stratigility\EmptyPipelineHandler))
> #15 .../Tinebase/Server/Expressive.php(106): Zend\Stratigility\MiddlewarePipe->handle(Object(Zend\Diactoros\ServerRequest))
> #16 .../Tinebase/Core.php(267): Tinebase_Server_Expressive->handle(Object(Tinebase_Http_Request))
> #17 .../index.php(24): Tinebase_Core::dispatchRequest()
> #18 {main}

If I create the following rewrite rule in /www/.htaccess, an autodiscover works with the help of an PHP script.
```
# Autodiscover
RewriteRule ^Autodiscover/Autodiscover.xml  autodiscover.php
```

Where is my mistake?

Thanks,
grEEtZ

**Comment by lab-at-nohl** [comment]
Hi, 

thanks for the very good, almost perfect description. Furthermore, a log file step DEBUG would be helpful. 

You can see that Tine 2.0 catches the request. 

> `#2 .../Tinebase/Expressive/RouteHandler.php(185): Tinebase_Controller->publicApiMSAutodiscoverXml()`

The section `# Routing` (under base directory) passes all files/directories not found to the index.php in the subdirectory tine. This might be intended, but will process every missing file by Tine 2.0. Which might cause trouble... 

I can only guess because running in a subfolder is not the default behaviour. You could e.g. try to have an additional rule between the existing two inside `# Routing` like this: 

`RewriteRule ^Autodiscover/Autodiscover.xml  /tine/index.php?doRouting=1 [E=REMOTE_USER:%{HTTP:Authorization},L,QSA]`

Best regards 
Johannes
